kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/FeatureService.java

```diff
@@ -39,23 +39,25 @@
       features.add(Mono.just(Feature.SCHEMA_REGISTRY));
     }
 
-    features.add(
-        isTopicDeletionEnabled(cluster)
-            .flatMap(r -> r ? Mono.just(Feature.TOPIC_DELETION) : Mono.empty())
-    );
+    if (controller != null) {
+      features.add(
+          isTopicDeletionEnabled(cluster, controller)
+              .flatMap(r -> r ? Mono.just(Feature.TOPIC_DELETION) : Mono.empty())
+      );
+    }
 
     return Flux.fromIterable(features).flatMap(m -> m).collectList();
   }
 
-  private Mono<Boolean> isTopicDeletionEnabled(KafkaCluster cluster) {
-    return brokerService.getController(cluster)
-        .map(Node::id)
-        .flatMap(broker -> brokerService.getBrokerConfigMap(cluster, broker))
-        .map(config -> {
-          if (config != null && config.get(DELETE_TOPIC_ENABLE) != null) {
-            return Boolean.parseBoolean(config.get(DELETE_TOPIC_ENABLE).getValue());
-          }
-          return false;
-        });
+  private Mono<Boolean> isTopicDeletionEnabled(KafkaCluster cluster, Node controller) {
+    return adminClientService.get(cluster)
+        .flatMap(ac -> ac.loadBrokersConfig(List.of(controller.id())))
+        .map(config ->
+            config.values().stream()
+                .flatMap(Collection::stream)
+                .filter(e -> e.name().equals("delete.topic.enable"))
```

Comment:
Why are we ditching a constant (`DELETE_TOPIC_ENABLE`) in favor of a magic number?

Reply:
I personally not a fan of globally-spread constants when they are used in one place. I create a constant in FeatureService

Reply:
thanks